test/jest/user-utils.spec.ts

```diff
@@ -1,50 +1,139 @@
+import {
+    IMParticleUser,
+    IdentityModifyResultBody,
+    IdentityResultBody,
+} from '../../src/identity-user-interfaces';
+import { SDKIdentityTypeEnum } from '../../src/identity.interfaces';
+import {
+    hasMPIDAndUserLoginChanged,
+    hasMPIDChanged,
+} from '../../src/user-utils';
 
 describe('user-utils', () => {
     describe('#didUserChange', () => {
         it('returns true if previousUser is null', () => {
-            expect(didUserChange(null, null)).toBeTruthy();
+            expect(hasMPIDAndUserLoginChanged(null, null)).toBeTruthy();
         });
 
         it('returns true if previousUser and newUser have different mpids', () => {
-            const previousUser = {
+            const previousUser = ({
                 getMPID: () => '123',
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            const newUser = {
+            const newUser = ({
                 getMPID: () => '456',
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            expect(didUserChange(previousUser, newUser)).toBeTruthy();
+            expect(
+                hasMPIDAndUserLoginChanged(previousUser, newUser)
+            ).toBeTruthy();
         });
 
         it('returns false if previousUser and newUser have the same MPID', () => {
-            const previousUser = {
+            const previousUser = ({
                 getMPID: () => '123',
                 isLoggedIn: () => false,
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            const newUser = {
+            const newUser = ({
                 getMPID: () => '123',
                 isLoggedIn: () => false,
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            expect(didUserChange(previousUser, newUser)).toBeFalsy();
+            expect(
+                hasMPIDAndUserLoginChanged(previousUser, newUser)
+            ).toBeFalsy();
         });
 
         it('returns true if previousUser and newUser have the same MPID but different login states', () => {
-            const previousUser = {
+            const previousUser = ({
                 getMPID: () => '123',
                 isLoggedIn: () => true,
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            const newUser = {
+            const newUser = ({
                 getMPID: () => '123',
                 isLoggedIn: () => false,
-            } as unknown as IMParticleUser;
+            } as unknown) as IMParticleUser;
 
-            expect(didUserChange(previousUser, newUser)).toBeTruthy();
+            expect(
+                hasMPIDAndUserLoginChanged(previousUser, newUser)
+            ).toBeTruthy();
+        });
+    });
+
+    describe('#hasMPIDChanged', () => {
+        const identityResultBody: IdentityResultBody = {
```

Comment:
there are a few tests here where the incoming user is empty, or mpid is null/undefined.  this makes sense from a code perspective, but maybe there needs to be a jira ticket added about determining which is not actually possible from a use case perspective.

Reply:
Created https://go.mparticle.com/work/SQDSDKS-6504 to address this.